chimney-macro-commons/src/main/scala-3/io/scalaland/chimney/internal/compiletime/datatypes/ValueClassesPlatform.scala

```diff
@@ -7,44 +7,47 @@ trait ValueClassesPlatform extends ValueClasses { this: DefinitionsPlatform =>
   import quotes.*, quotes.reflect.*
   import Type.platformSpecific.*
 
-  protected object ValueClassType extends ValueClassTypeModule {
+  protected object WrapperClassType extends WrapperClassTypeModule {
 
-    def parse[A: Type]: Option[Existential[ValueClass[A, *]]] = if Type[A].isAnyVal && !Type[A].isPrimitive then {
+    def parse[A: Type]: Option[Existential[WrapperClass[A, *]]] = {
       val A: TypeRepr = TypeRepr.of[A]
       val sym: Symbol = A.typeSymbol
 
-      val getter: Symbol = sym.declarations.headOption.getOrElse {
-        assertionFailed(s"AnyVal ${Type.prettyPrint[A]} expected to have 1 parameter")
+      val getterOpt: Option[Symbol] = sym.declarations.headOption
+      val primaryConstructorOpt: Option[Symbol] = Option(sym.primaryConstructor).filter(_.isClassConstructor)
+      val argumentOpt: Option[Symbol] = primaryConstructorOpt.flatMap { primaryConstructor =>
+        paramListsOf(primaryConstructor).flatten match {
+          case argument :: Nil => Some(argument)
+          case _               => None
+        }
       }
 
-      val primaryConstructor: Symbol = Option(sym.primaryConstructor).filter(_.isClassConstructor).getOrElse {
-        assertionFailed(s"AnyVal ${Type.prettyPrint[A]} expected to have 1 public constructor")
+      (getterOpt, primaryConstructorOpt, argumentOpt) match {
+        case (Some(getter), Some(primaryConstructor), Some(argument)) if !Type[A].isPrimitive =>
+          val argumentT = paramsWithTypes(A, primaryConstructor)(argument.name).asType.asInstanceOf[Type[Any]]
+          val inner = returnTypeOf[Any](A.memberType(getter)).asExistential
+          import inner.Underlying as Inner
+          assert(
+            argumentT =:= Inner,
```

Comment:
Wondering if we should check this by assertion or with if-else. I'm worried it can be hacked by the user by providing specially prepared class, something like:

```scala
class Foo(private val b: Int) {
  def a: String = b.toString
}
```